prow/crier/controller.go

```diff
@@ -92,11 +92,17 @@ func (r *reconciler) updateReportState(pj *prowv1.ProwJob, log *logrus.Entry, re
 		return fmt.Errorf("failed to patch: %w", err)
 	}
 
+	name := types.NamespacedName{Namespace: pj.Namespace, Name: pj.Name}
 	// Block until the update is in the lister to make sure that events from another controller
 	// that also does reporting dont trigger another report because our lister doesn't yet contain
 	// the updated Status
-	name := types.NamespacedName{Namespace: pj.Namespace, Name: pj.Name}
-	if err := wait.Poll(time.Second, 3*time.Second, func() (bool, error) {
+	backoff := wait.Backoff{
+		Duration: 100 * time.Millisecond,
```

Comment:
Is 100ms a reasonable time to expect to have already received the update to the watch?

Reply:
I don't have any data on this, but I'd expect so.  The check should be very cheap anyways and optimizing for reducing the locking time seems more important than reducing the number of cheap attempts. 
If you'd like I could add some debug logs as well to indicate how often we have to retry so that we can tweak these parameters, but I think these values will work just fine.

Reply:
Not a very strong opinion one way or another, but we could maybe just stick with the `wait.Poll` but do it every 100ms? Since this just attempts to get data from the cache its a super cheap thing to do

Reply:
I usually prefer exponential backoff, but that seems reasonable here since this is so cheap and since there will only be one thread in a polling loop per controller. That should also reduce the average amount of time we spend unnecessarily waiting compared to exponential backoff which is important because this blocks the controller. I'll close this PR and open another to simply decrease the poll period.
/close